sdk/core/Azure.Core/tests/Base64UrlTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Microsoft.Identity.Client;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class Base64UrlTests
+    {
+        [Test]
+        public void ValidateEncodeDecodeZeroLength()
+        {
+            var data = new byte[0];
+
+            var encoded = Base64Url.Encode(data);
+
+            Assert.AreEqual(string.Empty, encoded);
+
+            var decoded = Base64Url.Decode(encoded);
+
+            CollectionAssert.AreEqual(data, decoded);
+        }
+
+        [Test]
+        public void ValidateEncodeDecodeStringZeroLength()
+        {
+            var data = string.Empty;
+
+            var encoded = Base64Url.EncodeString(data);
+
+            Assert.AreEqual(string.Empty, encoded);
+
+            var decoded = Base64Url.DecodeString(encoded);
+
+            Assert.AreEqual(data, decoded);
+        }
+
+        [Test]
+        public void ValidateEncodeDecodeRandom()
+        {
+            var seed = new Random().Next();
+
+            var rand = new Random(seed);
+
+            for (int i = 1; i <= 512; i++)
+            {
+                for (int j = 0; j < 32; j++)
+                {
+                    var data = new byte[i];
+
+                    rand.NextBytes(data);
+
+                    var encoded = Base64Url.Encode(data);
+
+                    var decoded = Base64Url.Decode(encoded);
+
+                    CollectionAssert.AreEqual(data, decoded, "Data round trip failed. Seed {0}", seed);
```

Comment:
There should be a test against a hardcoded value as well. Here, you're making sure that encode and decode are congruent. That's great, you're checking for consistency. However, it's not checking for accuracy apart from the empty buffer case above.

Reply:
added.